// react-native-drawer-layout/lib/module/views/GestureHandler.js
import * as React from 'react';
import { View } from 'react-native';

// FIXME: Inline this type instead of getting it from react-native-gesture-handler
// Otherwise, we get a type error:
// Exported variable 'GestureDetector' has or is using name 'GestureDetectorProps' from external module ".." but cannot be named.
export const GestureDetector = _ref => {
  let {
    gesture: _0,
    userSelect: _1,
    children
  } = _ref;
  return /*#__PURE__*/React.createElement(React.Fragment, null, children);
};
export const GestureHandlerRootView = View;
export const Gesture = undefined;
export let GestureState = /*#__PURE__*/function (GestureState) {
  GestureState[GestureState["UNDETERMINED"] = 0] = "UNDETERMINED";
  GestureState[GestureState["FAILED"] = 1] = "FAILED";
  GestureState[GestureState["BEGAN"] = 2] = "BEGAN";
  GestureState[GestureState["CANCELLED"] = 3] = "CANCELLED";
  GestureState[GestureState["ACTIVE"] = 4] = "ACTIVE";
  GestureState[GestureState["END"] = 5] = "END";
  return GestureState;
}({});
